### PR TITLE
AGENT-18: Configure static network addresses from manifest file

### PR DESCRIPTION
**Pull request by bfournie**
Configure static network addresses on the node using the nmstateconfig manifest file. This uses the assisted-service staticnetworkconfig pkg, the version of assisted-installer was updated to pick up the most recent changes in staticnetworkconfig.

One caveat - 'nmstatectl' must be installed on the machine running 'fleeting' since assisted-service uses it for the gen-code command. This will change shortly when the changes to add 'nmstatectl gc' merge to https://github.com/nmstate/nmstate and are used by assisted-service. Currently only setting the address for Node 0 is supported.

This builds on the existing demo using this manifest file https://gist.github.com/bfournie/ad0fd2e827d028736530df0c857a164b, the IP in the manifest must also be added to the libvirt DNS configuration. The install will not complete with a valid hostname.

Follow-up PRs will address:
- Unit tests
- Getting the node zeto IP address from the nmstateconfig manifest file and removing the need to set this address in the CLI
- Supporting multiple hosts in one nmstateconfig file
- Getting the hostname from a install yaml file and setting it on host

### CODE REVIEW

File: pkg/manifests/nmstateconfig.go [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 )
 
-func getNMStateConfig() aiv1beta1.NMStateConfig {
+func getNMStateConfig() (aiv1beta1.NMStateConfig, error) {
 	var nmStateConfig aiv1beta1.NMStateConfig
 	if err := GetFileData("nmstateconfig.yaml", &nmStateConfig); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+		// This file is optional if DHCP addressing is used
+		return nmStateConfig, err
```
Comment (celebdor) [inline pkg/manifests/nmstateconfig.go:19]: This is temporary, right? We won't support fully dhcp envs any time soon

Comment (bfournie) [reply pkg/manifests/nmstateconfig.go:19]: Yes, as we talked about, only temporary until https://github.com/openshift-agent-team/fleeting/pull/22 merges

**Comment by pawanpinjarkar** [comment]
Tested the changes from this PR from dev-scripts https://github.com/openshift-metal3/dev-scripts/pull/1380 and /lgtm

**Pull request merged**